resource-managers/yarn/src/main/resources/META-INF/services/org.apache.spark.deploy.security.ServiceCredentialProvider

```diff
@@ -0,0 +1,3 @@
+org.apache.spark.deploy.yarn.security.YARNHadoopFSCredentialProvider
```

Comment:
Oh yea, I guess the standard Spark distributions are compiled with both profiles.  I created a separate `YARNHadoopFSCredentialProvider` to handle YARN's specific functionality (renewer, extra filesystems, etc.), but I guess I could do that through parameterization rather than a separate class, since it seems we can't have a YARN specific class through service loading.  How does that sound?